src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -604,11 +633,33 @@ public void cutConnectivity(LfNetwork lfNetwork, GraphDecrementalConnectivity<Lf
     protected void setPredefinedResults(Collection<LfSensitivityFactor<V, E>> lfFactors, Set<LfBus> connectedComponent,
                                         GraphDecrementalConnectivity<LfBus> connectivity) {
         for (LfSensitivityFactor<V, E> factor : lfFactors) {
-            // check if the factor function and variable are in different connected components
-            if (factor.areVariableAndFunctionDisconnected(connectivity)) {
-                factor.setPredefinedResult(0d);
-            } else if (!factor.isConnectedToComponent(connectedComponent)) {
-                factor.setPredefinedResult(Double.NaN); // works for sensitivity and function reference
+            if (factor.getStatus() == LfSensitivityFactor.Status.VALID) {
+                // after a contingency, we check if the factor function and the variable are in different connected components
+                boolean variableConnected = factor.isVariableConnectedToSlackComponent(connectedComponent);
+                boolean functionConnected = factor.isFunctionConnectedToSlackComponent(connectedComponent);
+                if (!variableConnected && functionConnected) {
+                    // VALID_ONLY_FOR_FUNCTION status
+                    factor.setSensitivityValuePredefinedResult(0d);
+                }
+                if (!variableConnected && !functionConnected) {
+                    // SKIP status
```

Comment:
Here I think that this level of details is not needed: in case of factor with SKIP status, in pre-contingency state, you write `NaN` for sensi and `NaN` for function. But you introduce a new level of detail after a contingency. It is not coherent and I think that if it is more right, it is less perfomant.

Reply:
I agree with you.
Indeed, in precontingency network, skip status is used when both variable and function are disconnected from slack and the sensitivity is set to NaN.
But in case of a contingency breaking connectivity we treat the case where both function and element are disconnected from slack and are in different connected components with a 0 value for the sensitivity.

This behavior is not coherent.